src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js

```diff
@@ -0,0 +1,317 @@
+*/
+
+import { describe, it, expect, vi } from "vitest";
+import { mount } from "@vue/test-utils";
+import Signup from "../Signup.vue";
+import Login from "../Login.vue";
+import Logo from "../Logo.vue";
+import LoggedOut from "../LoggedOut.vue";
+import ModalFullScreen from "./../../ModalFullScreen.vue";
+// import axios from "axios";
+import { createTestingPinia } from "@pinia/testing";
+
+describe("Login.vue", () => {
+  // Check if login exists
+  it("Renders with null prop", () => {
+    const componentWrapper = mount(Login, {
+      props: {
+        /*
+        TODO(Add API variables here)
+        */
+      },
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    expect(componentWrapper.exists()).toBe(true);
+  });
+
+  it("Check if the login_components_container exists", async () => {
+    const componentWrapper = mount(Login, {
+      props: {},
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const login_components_container = componentWrapper.find(
+      ".login_components_container"
+    );
+    expect(login_components_container.exists()).toBe(true);
+  });
+
+  it("Check if the loginForm works", async () => {
+    const componentWrapper = mount(Login, {
+      props: {},
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const loginForm = componentWrapper.find("#loginForm");
+    expect(loginForm.exists()).toBe(true);
+    await loginForm.trigger("submit.prevent");
+  });
+
+  it("Check if the loginThroughSSO works", async () => {
+    const componentWrapper = mount(Login, {
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const loginWithGoogle = componentWrapper.find("#loginWithGoogle");
+    expect(loginWithGoogle.exists()).toBe(true);
+    await loginWithGoogle.trigger("submit.prevent");
+  });
+
+  it("Check if the switchToSignupButton works", async () => {
+    const componentWrapper = mount(Login, {
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const switchToSignupButton = componentWrapper.find("#switchToSignupButton");
+    expect(switchToSignupButton.exists()).toBe(true);
+    await switchToSignupButton.trigger("click");
+  });
+
+  it("Check if the small titles work", async () => {
+    const componentWrapper = mount(Login, {
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const smallTitles = componentWrapper.find("#small");
+    expect(smallTitles.exists()).toBe(true);
+  });
+
+  it("Check if the big titles work", async () => {
+    const componentWrapper = mount(Login, {
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const bigTitles = componentWrapper.find("#big");
+    expect(bigTitles.exists()).toBe(true);
```

Comment:
I don't see all these test's purpose, if you are testing all these component works, the first test, run component with no prop should do it, why testing each and single title here?

Reply:
I am not sure what you meant. I have seen component tests on the previous file before. But if your case is true, then this is the comment that would have been very beneficial on day one

Reply:
Would you let me know where you have seen a <div> is checked in previous test, I'd take a look